cmd/disk-cache.go

```diff
@@ -0,0 +1,764 @@
+		info, err = putObjectPartFn(bucket, object, uploadID, partID, hashReader)
+		if err != nil {
+			close(pinfoCh)
+			pipeWriter.CloseWithError(err)
+			return
+		}
+		pinfoCh <- info
+	}()
+	go func() {
+		if _, perr := dcache.PutObjectPart(bucket, object, uploadID, partID, cHashReader); perr != nil {
+			wPipe.CloseWithError(perr)
+			return
+		}
+	}()
+
+	mwriter := io.MultiWriter(pipeWriter, wPipe)
+	_, err = io.Copy(mwriter, data)
+	if err != nil {
+		pipeWriter.CloseWithError(err)
+		wPipe.CloseWithError(err)
+		return PartInfo{}, err
+	}
+	info = <-pinfoCh
+	return info, err
+}
+
+// AbortMultipartUpload - aborts multipart upload on backend and cache.
+func (c cacheObjects) AbortMultipartUpload(bucket, object, uploadID string) error {
+	abortMultipartUploadFn := c.AbortMultipartUploadFn
+
+	if c.isCacheExclude(bucket, object) {
+		return abortMultipartUploadFn(bucket, object, uploadID)
+	}
+
+	dcache, err := c.cache.getCacheFS(bucket, object, false)
+	if err != nil {
+		// disk cache could not be located,execute backend call.
+		return abortMultipartUploadFn(bucket, object, uploadID)
+	}
+	// execute backend operation
+	err = abortMultipartUploadFn(bucket, object, uploadID)
+	if err != nil {
+		return err
+	}
+	// abort multipart upload on cache
+	dcache.AbortMultipartUpload(bucket, object, uploadID)
+	return nil
+}
+
+// CompleteMultipartUpload - completes multipart upload operation on backend and cache.
+func (c cacheObjects) CompleteMultipartUpload(bucket, object, uploadID string, uploadedParts []CompletePart) (objInfo ObjectInfo, err error) {
+	completeMultipartUploadFn := c.CompleteMultipartUploadFn
+
+	if c.isCacheExclude(bucket, object) {
+		return completeMultipartUploadFn(bucket, object, uploadID, uploadedParts)
+	}
+
+	dcache, err := c.cache.getCacheFS(bucket, object, false)
+	if err != nil {
+		// disk cache could not be located,execute backend call.
+		return completeMultipartUploadFn(bucket, object, uploadID, uploadedParts)
+	}
+	// perform backend operation
+	objInfo, err = completeMultipartUploadFn(bucket, object, uploadID, uploadedParts)
+	if err != nil {
+		return
+	}
+	// create new multipart upload in cache with same uploadID
+	dcache.CompleteMultipartUpload(bucket, object, uploadID, uploadedParts)
+	return
+}
+
+// initCaches initializes the cacheFSObjects for the "drives" specified in config.json
+// or the global env overrides.
+func initCaches(c CacheConfig) (*diskCache, error) {
+	var cfsObjects []*cacheFSObjects
+	for _, dir := range c.Drives {
+		if dir == "" {
+			continue
+		}
+		c, err := newCacheFSObjects(dir, c.Expiry)
+		if err != nil {
+			return nil, err
+		}
+		// Start the purging go-routine
+		go c.purge()
+		cfsObjects = append(cfsObjects, c)
+	}
+
+	if err := loadAndValidateCacheFormat(cfsObjects); err != nil {
+		errorIf(err, "Cache drives validation error")
+	}
+	return &diskCache{cfs: cfsObjects}, nil
+}
+
+// Return error if Atime is disabled on the O/S
+func checkAtimeSupport() (err error) {
```

Comment:
what needs to be done is, the file needs to be created on the cache-dirs, using os.TempDir() will use `/tmp`

Also if a partition is mounted with atime disabled, `os.Chtimes(file.Name(), atime, mtime)` will still succeed.

So, once you create a file note the atime using os.Stat(), then perform a Read() operation and do os.Stat() again to see if the Atime changed.